test/base/src/androidTest/kotlin/io/realm/shared/PrimaryKeyTests.kt

```diff
@@ -95,7 +96,7 @@ class PrimaryKeyTests {
         realm.writeBlocking {
             val obj = PrimaryKeyString().apply { primaryKey = PRIMARY_KEY }
             copyToRealm(obj)
-            assertFailsWith<RuntimeException> {
+            assertFailsWith<RealmPrimaryKeyConstraintException> {
```

Comment:
Again, seems weird that we have a specific public error for that instead of just using `IllegalArgumentException`. Maybe there is a historic reason, @cmelchior?